language/stdlib/modules/LibraAccount.move

```diff
@@ -1238,14 +1357,19 @@ module LibraAccount {
             sender_account.sequence_number < (MAX_U64 as u64),
             Errors::limit_exceeded(ESEQUENCE_NUMBER)
         );
+        assert(
+            sender_account.sequence_number == txn_sequence_number,
+            Errors::invalid_argument(ESEQUENCE_NUMBER)
```

Comment:
In the process of specifying the epilogue I wound up coming across this and wonder what connection should `txn_sequence_number` have w.r.t. the the account's sequence number? Should `txn_sequence_number` not be passed in and we bump it manually here? 

What are your thoughts @runtian-zhou  @tnowacki and @bob-wilson?

Reply:
My guess is that @sblackshear added this code? I don't know the history of it. If its not used now, I'd rather just not pass it in, but we should probably wait to hear from him

Reply:
Good question. I think the reasoning for passing in `txn_sequence_number` and using `txn_sequence_number + 1` at 1241 in the old code (1382 in the new code) is that it guarantees (in conjunction with the prologue) that the sequence number increases by exactly 1. If we don't pass it in and increment the account sequence number instead, it is possible that code in the script (or WriteSet!) already increased the sequence number (or rewrote it to a smaller value in the WriteSet case).

I think continuing to pass in `txn_sequence_number`, keeping this assertion, and also keeping the code at 1241 probably makes the most sense.